Libraries/LibWeb/CSS/Parser/PropertyParsing.cpp

```diff
@@ -394,7 +394,7 @@ Parser::ParseErrorOr<NonnullRefPtr<CSSStyleValue const>> Parser::parse_css_value
     if (property_id == PropertyID::Custom || contains_arbitrary_substitution_function)
         return UnresolvedStyleValue::create(move(component_values), contains_arbitrary_substitution_function, original_source_text);
 
-    if (component_values.is_empty())
+    if (component_values.is_empty() || unprocessed_tokens.has_next_token())
```

Comment:
Like this, we'd ignore the semicolon if we're creating an `UnresolvedStyleValue` above.

(My suggestion before was to put the check in `parse_as_css_value()` instead, because multiple places call this method and not all of them necessarily only pass in tokens for for the css value. Though it seems like that is actually true.)

So I'd suggest replacing the `break` on line 373 with just returning the syntax error directly.